src/randomizer.py

```diff
@@ -2080,6 +2099,78 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+def perm_inline_get_structmember(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Creates an inline function for accessing a struct member."""
+
+    typemap = build_typemap(ast)
+
+    cands: List[Union[ca.StructRef, ca.UnaryOp]] = []
+    for expr in get_block_expressions(fn.body, region):
+        if isinstance(expr, ca.StructRef) and expr.type == "->":
+            cands.append(expr)
+        if (
+            isinstance(expr, ca.UnaryOp)
+            and expr.op is "&"
+            and isinstance(expr.expr, ca.StructRef)
+            and expr.expr.type == "->"
+        ):
+            cands.append(expr)
+
+    ensure(cands)
+    cand = random.choice(cands)
+
+    parent_op: Optional[ca.UnaryOp] = None
+    if isinstance(cand, ca.UnaryOp):
+        parent_op = cand
+        assert isinstance(parent_op.expr, ca.StructRef)
+        cand = parent_op.expr
+
+    member_type: SimpleType = decayed_expr_type(cand, typemap)
+    member_name = cand.field.name
+
+    new_inline_fn_name = get_noncolliding_name(ast, f"get_{member_name}")
+
+    # Replace the StructRef with a FuncCall to the new inline yet to be created
+    replace_node(
+        fn.body,
+        parent_op or cand,
+        ca.FuncCall(ca.ID(new_inline_fn_name), ca.ExprList([cand.name])),
+    )
+
+    # Now create the new inline function definition
+    arg_type: SimpleType = decayed_expr_type(cand.name, typemap)
+
+    param = ast_util.make_decl("x", arg_type)
+
+    fn_decl = ast_util.make_decl(
```

Comment:
now that make_decl makes these so short we can imo inline them into the usage

Reply:
I inlined one of them into the usage